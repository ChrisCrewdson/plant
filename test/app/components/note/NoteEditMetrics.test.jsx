
const React = require('react');
const renderer = require('react-test-renderer');
const MuiThemeProvider = require('material-ui/styles/MuiThemeProvider').default;
const getMuiTheme = require('material-ui/styles/getMuiTheme').default;
const lightBaseTheme = require('material-ui/styles/baseThemes/lightBaseTheme').default;
const {
  MemoryRouter,
} = require('react-router-dom');
const { Provider } = require('react-redux');
const NoteEditMetrics = require('../../../../app/components/note/NoteEditMetrics');
const store = require('../../../../app/store');
const App = require('../../../../app/components/App');

const muiTheme = getMuiTheme(lightBaseTheme);

describe('NoteEditMetrics', () => {
  const storeDispatch = store.dispatch;

  beforeAll(() => {
    store.dispatch = () => {};
  });

  afterAll(() => {
    store.dispatch = storeDispatch;
  });

  test('should render NoteEditMetrics', () => {
    const interimNote = {
    };

    const component = renderer.create(
      <MuiThemeProvider muiTheme={muiTheme}>
        <Provider store={store}>
          <App>
            <MemoryRouter>
              <NoteEditMetrics
                dispatch={storeDispatch}
                interimNote={interimNote}
              />
            </MemoryRouter>
          </App>
        </Provider>
      </MuiThemeProvider>);
    const noteEditMetrics = component.toJSON();
    expect(noteEditMetrics).toMatchSnapshot();
  });
});
