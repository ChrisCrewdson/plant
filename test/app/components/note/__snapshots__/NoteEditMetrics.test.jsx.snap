// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteEditMetrics should render NoteEditMetrics 1`] = `
<div
  className="react-root"
>
  <div
    style={
      Object {
        "textAlign": "left",
      }
    }
  >
    <div>
      <div
        style={
          Object {
            "backgroundColor": "transparent",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 72,
            "lineHeight": "24px",
            "marginLeft": 20,
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      >
        <label
          htmlFor="height"
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.3)",
              "lineHeight": "22px",
              "muiPrepared": true,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 38,
              "transform": "scale(1) translate(0, 0)",
              "transformOrigin": "left top",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "zIndex": 1,
            }
          }
        >
          Height (inches only)
        </label>
        <div
          style={
            Object {
              "bottom": 12,
              "color": "rgba(0, 0, 0, 0.3)",
              "muiPrepared": true,
              "opacity": 0,
              "position": "absolute",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          Enter height of plant
        </div>
        <input
          disabled={false}
          id="height"
          name="height"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "WebkitOpacity": 1,
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "appearance": "textfield",
              "backgroundColor": "rgba(0,0,0,0)",
              "border": "none",
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "inherit",
              "font": "inherit",
              "height": "100%",
              "marginTop": 14,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
          type="number"
          value=""
        />
        <div>
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 1,
                "borderColor": "#e0e0e0",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "display": "none",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "width": "100%",
              }
            }
          />
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 2,
                "borderColor": "#00bcd4",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "display": "none",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "transform": "scaleX(0)",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "100%",
              }
            }
          />
        </div>
        
      </div>
      <div
        style={
          Object {
            "backgroundColor": "transparent",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 72,
            "lineHeight": "24px",
            "marginLeft": 20,
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      >
        <label
          htmlFor="girth"
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.3)",
              "lineHeight": "22px",
              "muiPrepared": true,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 38,
              "transform": "scale(1) translate(0, 0)",
              "transformOrigin": "left top",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "zIndex": 1,
            }
          }
        >
          Girth (inches only)
        </label>
        <div
          style={
            Object {
              "bottom": 12,
              "color": "rgba(0, 0, 0, 0.3)",
              "muiPrepared": true,
              "opacity": 0,
              "position": "absolute",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          Enter girth of plant
        </div>
        <input
          disabled={false}
          id="girth"
          name="girth"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "WebkitOpacity": 1,
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "appearance": "textfield",
              "backgroundColor": "rgba(0,0,0,0)",
              "border": "none",
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "inherit",
              "font": "inherit",
              "height": "100%",
              "marginTop": 14,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
          type="number"
          value=""
        />
        <div>
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 1,
                "borderColor": "#e0e0e0",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "display": "none",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "width": "100%",
              }
            }
          />
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 2,
                "borderColor": "#00bcd4",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "display": "none",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "transform": "scaleX(0)",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "100%",
              }
            }
          />
        </div>
        
      </div>
      <div
        style={
          Object {
            "backgroundColor": "transparent",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 72,
            "lineHeight": "24px",
            "marginLeft": 20,
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      >
        <label
          htmlFor="harvestCount"
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.3)",
              "lineHeight": "22px",
              "muiPrepared": true,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 38,
              "transform": "scale(1) translate(0, 0)",
              "transformOrigin": "left top",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "zIndex": 1,
            }
          }
        >
          Harvest Count
        </label>
        <div
          style={
            Object {
              "bottom": 12,
              "color": "rgba(0, 0, 0, 0.3)",
              "muiPrepared": true,
              "opacity": 0,
              "position": "absolute",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          Enter number of items harvested
        </div>
        <input
          disabled={false}
          id="harvestCount"
          name="harvestCount"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "WebkitOpacity": 1,
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "appearance": "textfield",
              "backgroundColor": "rgba(0,0,0,0)",
              "border": "none",
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "inherit",
              "font": "inherit",
              "height": "100%",
              "marginTop": 14,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
          type="number"
          value=""
        />
        <div>
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 1,
                "borderColor": "#e0e0e0",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "display": "none",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "width": "100%",
              }
            }
          />
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 2,
                "borderColor": "#00bcd4",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "display": "none",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "transform": "scaleX(0)",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "100%",
              }
            }
          />
        </div>
        
      </div>
      <div
        style={
          Object {
            "backgroundColor": "transparent",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 72,
            "lineHeight": "24px",
            "marginLeft": 20,
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      >
        <label
          htmlFor="harvestWeight"
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.3)",
              "lineHeight": "22px",
              "muiPrepared": true,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 38,
              "transform": "scale(1) translate(0, 0)",
              "transformOrigin": "left top",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "zIndex": 1,
            }
          }
        >
          Harvest Weight (lbs only)
        </label>
        <div
          style={
            Object {
              "bottom": 12,
              "color": "rgba(0, 0, 0, 0.3)",
              "muiPrepared": true,
              "opacity": 0,
              "position": "absolute",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          Enter weight of harvest
        </div>
        <input
          disabled={false}
          id="harvestWeight"
          name="harvestWeight"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "WebkitOpacity": 1,
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "appearance": "textfield",
              "backgroundColor": "rgba(0,0,0,0)",
              "border": "none",
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "inherit",
              "font": "inherit",
              "height": "100%",
              "marginTop": 14,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
          type="number"
          value=""
        />
        <div>
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 1,
                "borderColor": "#e0e0e0",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "display": "none",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "width": "100%",
              }
            }
          />
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 2,
                "borderColor": "#00bcd4",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "display": "none",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "transform": "scaleX(0)",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "100%",
              }
            }
          />
        </div>
        
      </div>
    </div>
  </div>
</div>
`;
